timeseries/src/autogluon/timeseries/models/chronos/pipeline/utils.py

```diff
@@ -1,14 +1,210 @@
+    compatible with Chronos models.
+
+    When a ``tokenizer`` is provided, data is converted into HuggingFace-compatible set of
+    ``input_ids``, ``attention_mask`` and ``labels``, used by the original Chronos models.
+
+    When the ``tokenizer`` is omitted, data is converted into the format compatible with
+    ChronosBolt models, i.e., ``context`` and ``target``.
+
+    Parameters
+    ----------
+    target_df : TimeSeriesDataFrame
+        The ``TimeSeriesDataFrame`` to be converted
+    target_column : str, default = "target"
+        The name of the column which contains the target time series, by default "target"
+    context_length : int, default = 512
+        The length of the historical context
+    prediction_length : int, default = 64
+        The prediction_length, i.e., length of label or target
+    tokenizer : ``ChronosTokenizer``, default = None
+        When a ``ChronosTokenizer`` object is provided, data will be converted into the
+        HuggingFace format accepted by the original Chronos models using this ``ChronosTokenizer``.
+        If None, data will be converted into the format accepted by ChronosBolt models.
+    mode : Literal["training", "validation"], default = "training"
+        When ``training``, random slices from the time series will be returned for training purposes.
+        If ``validation``, the last slice of each time series returned in the original order.
+    """
+
+    def __init__(
+        self,
+        target_df: TimeSeriesDataFrame,
+        target_column: str = "target",
+        context_length: int = 512,
+        prediction_length: int = 64,
+        tokenizer: Optional["ChronosTokenizer"] = None,
+        mode: Literal["training", "validation"] = "training",
+    ) -> None:
+        super().__init__()
+
+        assert mode in ("training", "validation")
+
+        # A dummy hourly freq is used because the model doesn't actually need the freq
+        self.gluonts_dataset = SimpleGluonTSDataset(target_df=target_df, freq="h", target_column=target_column)
+        self.tokenizer = tokenizer
+        self.context_length = context_length
+        self.prediction_length = prediction_length
+        self.mode = mode
+
+    def _create_instance_splitter(self, mode: str):
+        instance_sampler = {
+            "training": ExpectedNumInstanceSampler(num_instances=1.0, min_future=self.prediction_length),
+            "validation": ValidationSplitSampler(min_future=self.prediction_length),
+        }[mode]
+
+        return InstanceSplitter(
+            target_field=FieldName.TARGET,
+            is_pad_field=FieldName.IS_PAD,
+            start_field=FieldName.START,
+            forecast_start_field=FieldName.FORECAST_START,
+            instance_sampler=instance_sampler,
+            past_length=self.context_length,
+            future_length=self.prediction_length,
+            dummy_value=np.nan,
+        )
+
+    def _infinite_iterable(self, data: Iterable[dict]):
```

Comment:
nit: using `itertools.cycle` saves a bit of code and makes the intent clearer.

Reply:
`itertools.cycle` caches the data which we don't want here.

Reply:
then something like `chain.from_iterable(cycle([data]))` is still cleaner. this defines a protected instance method for this simple functionality. there is nothing in this definition that is associated with the class or an instance, or anything to suggest this can be the case in the future. it's not shared across methods, or doesn't serve to break up a method...